snuba/snuba_migrations/generic_metrics/0036_gauges_meta_tables.py

```diff
@@ -0,0 +1,182 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import AggregateFunction, Column, DateTime, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget
+from snuba.utils.schemas import Float
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    granularity = "8192"
+    meta_view_name = "generic_metric_gauges_meta_mv"
+    meta_local_table_name = "generic_metric_gauges_meta_local"
```

Comment:
Not a blocker. But is it necessary to build 2 tables? Wouldn't the `meta_tag_values` table also be able to answer the same questions about give back the list of all metric id's and tag keys?

Reply:
Not in it's current form, since I stripped `use_case_id` and `org_id` out of the table. Both of those would be needed to guarantee finding accurate `metric_id`.

Here was my thought process:

The table that satisfies finding metric IDs and tag keys needs `org_id` and `use_case_id`. However it doesn't need tag values, and tag values account for 98% of the size of that table (3.61GiB out of 3.66GiB).

The tag value table doesn't need `org_id` or `use_case_id`, since it is guaranteed to always have a `metric_id`. So by discarding those columns, I save a bunch of unnecessary duplicated storage.

So by having two tables, I can store the smallest amount of things, with as little duplicated data as possible.